packages/jest-environment-puppeteer/src/PuppeteerEnvironment.js

```diff
@@ -127,9 +127,48 @@ class PuppeteerEnvironment extends NodeEnvironment {
         }
         await this.global.jestPuppeteer.resetPage()
       },
-    }
+      keepTabOpened: async () => {
+        if (this.global.page) {
+          this.global.page.removeListener('pageerror', handleError)
+        }
+        if (config.browserContext === 'incognito' && this.global.context) {
+          await this.global.context.close()
+        } else if (this.global.page) {
+          await this.global.page.close()
+        }
+        this.global.page = null
+
+        if (this.global.browser) {
+          await this.global.browser.disconnect()
+        }
+
+        this.global.browser = await puppeteer.connect({
+          ...config.connect,
+          ...config.launch,
+          browserURL: undefined,
+          browserWSEndpoint: wsEndpoint,
+        });
 
-    await this.global.jestPuppeteer.resetBrowser()
+        if (config.browserContext === 'incognito') {
+          this.global.context = await this.global.browser.createIncognitoBrowserContext();
+          await this.global.jestPuppeteer.resetPage()
+        } else
+          if (config.browserContext === 'default' || !config.browserContext) {
+            this.global.context = await this.global.browser.browserContexts()[0];
```

Comment:
Why await?

Reply:
Await is unnecessary. I may have kept it initially for testing purposes.